google/cloud/storage/testbench/gcs_bucket.py

```diff
@@ -98,16 +99,28 @@ def apply_patch(self, patch):
         :param patch:dict a dictionary with metadata changes.
         """
         writeable_keys = {
-            'acl', 'billing', 'cors', 'defaultObjectAcl', 'encryption',
-            'labels', 'lifecycle', 'location', 'logging', 'storageClass',
-            'versioning', 'website'
+            'acl', 'billing', 'cors', 'defaultEventBasedHold',
+            'defaultObjectAcl', 'encryption', 'labels', 'lifecycle', 'location',
+            'logging', 'retentionPolicy', 'storageClass', 'versioning',
+            'website'
         }
         for key, value in patch.iteritems():
             if key not in writeable_keys:
                 raise error_response.ErrorResponse(
                     'Invalid metadata change. %s is not writeable' % key,
-                    status_code=503)
+                    status_code=412)
+        retention_policy = patch.get('retentionPolicy')
+        if retention_policy is not None:
+            for field in ['isLocked', 'effectiveTime']:
+                if retention_policy.get(field) is not None:
+                    raise error_response.ErrorResponse(
+                        'Invalid metadata change. %s is not writeable' % field,
```

Comment:
As mentioned in the other PR, the service actually just ignores these unwriteable fields if you try to set them in a PATCH request.  If it makes testing easier, I can deal with having the testbench fail instead, but note that this approach deviates from how the service actually responds.

Reply:
Fixed.